src/cmd/gz.cc

```diff
@@ -37,47 +40,119 @@
 #include "gz/sim/Server.hh"
 #include "gz/sim/ServerConfig.hh"
 
+#include "gz/sim/Util.hh"
+
+#ifdef WITH_GUI
 #include "gz/sim/gui/Gui.hh"
+#endif
 
 using namespace gz;
 
 //////////////////////////////////////////////////
-extern "C" char *gzSimVersion()
+void cmdVerbosity(const int _verbosity)
 {
-  return strdup(GZ_SIM_VERSION_FULL);
+  common::Console::SetVerbosity(_verbosity);
+
+  // SDFormat only has 2 levels: quiet / loud. Let sim users suppress all SDF
+  // console output with zero verbosity.
+  if (_verbosity == 0)
+  {
+    sdf::Console::Instance()->SetQuiet(true);
+  }
 }
 
 //////////////////////////////////////////////////
-extern "C" char *simVersionHeader()
+const std::string worldInstallDir()
 {
-  return strdup(GZ_SIM_VERSION_HEADER);
+  static std::string worldInstallDir = gz::sim::getWorldInstallDir();
+  return worldInstallDir;
 }
 
 //////////////////////////////////////////////////
-extern "C" void cmdVerbosity(
-    const char *_verbosity)
+int checkFile(std::string &_file)
 {
-  int verbosity = std::atoi(_verbosity);
-  common::Console::SetVerbosity(verbosity);
-
-  // SDFormat only has 2 levels: quiet / loud. Let sim users suppress all SDF
-  // console output with zero verbosity.
-  if (verbosity == 0)
+  // Check if passed string is valid
+  if(!_file.empty())
   {
-    sdf::Console::Instance()->SetQuiet(true);
+    std::string filepath;
+
+    // Check if passed file exists
+    if(std::filesystem::exists(_file))
+    {
+      filepath = _file;
+    }
+    else
+    {
+      // If passed file does not exist, check GZ_SIM_RESOURCE_PATH
+      // environment variable
+      auto resourcePaths = sim::resourcePaths();
+      for(auto path : resourcePaths)
+      {
+        std::string resourceFilepath =
+          common::joinPaths(path, _file);
+        if(std::filesystem::exists(resourceFilepath))
+        {
+          filepath = resourceFilepath;
+          break;
+        }
+      }
+
+      // If file does not exist in resource path, check in the
+      // installation location of available worlds
+      if(filepath.empty())
+      {
+        std::string worldPath =
+          common::joinPaths(worldInstallDir(), _file);
+        if(std::filesystem::exists(worldPath))
+        {
+          filepath = worldPath;
+        }
+        else
+        {
+          // Check Fuel for file
+          filepath = findFuelResource(_file);
+        }
+      }
+    }
+
+    if(filepath.empty())
+    {
+      std::cout << "Unable to find or download file "
+                << _file << std::endl;
+      return -1;
+    }
+
+    _file = filepath;
   }
+
+  return 0;
 }
 
 //////////////////////////////////////////////////
-extern "C" const char *worldInstallDir()
+int parseSdfFile(const std::string &_file,
+    std::string &_parsedSdfFile)
 {
-  static std::string worldInstallDir = gz::sim::getWorldInstallDir();
-  return worldInstallDir.c_str();
+  if(!_parsedSdfFile.empty())
```

Comment:
I don't understand this check. It looks like an empty `std::string` is passed to this function at `src/cmd/sim_main.cc:450-451` and `src/cmd/sim_main.cc:507-508`, but this function is a no-op for empty strings

Reply:
affected by https://github.com/gazebosim/gz-sim/pull/2849#discussion_r2089324195